concrete/controllers/search/file_folder.php

```diff
@@ -54,38 +56,46 @@ public function search(Query $query = null)
         }
 
         if (isset($folder)) {
-
             if ($folder instanceof SearchPreset) {
                 $search = $folder->getSavedSearchObject();
                 $query = $search->getQuery();
-                $provider = \Core::make('Concrete\Core\File\Search\SearchProvider');
+                $provider = $app->make(SearchProvider::class);
                 $ilr = $provider->getSearchResultFromQuery($query);
-                $ilr->setBaseURL(\URL::to('/ccm/system/search/files/preset', $search->getID()));
+                $ilr->setBaseURL(URL::to('/ccm/system/search/files/preset', $search->getID()));
             }
 
             $searchRequest->addToSearchRequest('folder', $folder->getTreeNodeID());
-
         }
 
         if (!isset($ilr)) {
-
             if (!isset($folder)) {
                 $folder = $this->filesystem->getRootFolder();
             }
             $u = new \User();
             $list = $folder->getFolderItemList($u, $this->request);
             $fields = $this->request->get('field');
-            $filters = array();
+            $filters = [];
             if (is_array($fields) && count($fields) > 0) { // We are passing in something like "filter by images"
                 $manager = ManagerFactory::get('file_folder');
                 $filters = $manager->getFieldsFromRequest($this->request->query->all());
             }
 
+            $provider = $app->make(SearchProvider::class);
+            $itemsPerPage = (int) $this->request->get('fSearchItemsPerPage');
```

Comment:
Should this value be maxed out against the max value from `concrete.file_manager.items_per_page_options`?